### PR TITLE
Add Sendable conformance

### PR DESCRIPTION
**Pull request by amyoulton**
This PR adds Sendable conformance to both Netable itself, as well as the example project.

## Netable

The changes in Netable were  _mostly simple_, with a few caveats. Every public store property within Netable has to conform to Sendable, as it's an actor. The most complicated of them is `AnyPublisher` and I have left a comment directly on that piece of code. In general, the Combine framework and Sendable are not a super big fan of one another. I'll discuss how this was mostly navigated lower down in the PR, but AnyPublisher threw the biggest wrench for sure.

We also previously discussed removing Sendable conformance from the `requestFailureDelegate` because it doesn't _need_ to be Sendable given how it operates, but it _does_ need to be Sendable given that it's a stored property within the Netable actor. 

## Example Project

Mostly of the issues were solved with declaring the code run on the MainActor, as discussed in [this slack](https://steamclock.slack.com/archives/C0WL7G3JL/p1674169801945799) thread. 

The only "odd ball" is the error service. I typically wouldn't want any service to be forced to run on the main thread, but given the nature of the error service in our case, I think it makes sense.

### CODE REVIEW

File: Netable/Netable/Netable.swift [hunk 1]
```diff
@@ -275,3 +275,5 @@ public actor Netable {
         }
     }
 }
+
+extension AnyPublisher: @unchecked Sendable {}
```
Comment (amyoulton) [inline Netable/Netable/Netable.swift:279]: I really didn't want to use `@unchecked` here, but if you're implementing it retroactively, you have to. :( I'm not sure if maybe there is a better way to tackle this?

**Comment by brendanlensink** [comment]
It looks like we've still got a whole bunch of Concurrency related warnings when building this project.

Are we able to resolve some of these, or is there some reason we can't deal with them?

**Comment by amyoulton** [comment]
The main project and both the Example project and Tests targets have complete currency checking selected but the netable target was apparently only marked minimal. I have changed this over and will fix this warnings.

**Review by brendanlensink — APPROVED** [review]
🚀

**Pull request merged**